### PR TITLE
Remove implicit test session creation

### PR DESCRIPTION
**Pull request by ninjinkun**
## Summary

This PR removes the implicit test session creation functionality from the Smart Tests CLI, streamlining the session management and removing unnecessary complexity.

## For reviewer

This PR includes important changes. Please check the following point.

**⚠️ Critical Review Point**: Please pay special attention to the `builds/{}/test_session_names/{}` endpoint functionality. We need to verify that this endpoint works correctly when `no_build` is specified, as the session creation logic has been significantly modified. The changes may affect how the API handles build-less test sessions.

Specifically, please review:
1. How `builds/{}/test_session_names/{}` behaves with `no_build` parameter
2. Whether the session name resolution works correctly without implicit session creation
3. If there are any edge cases where the endpoint might fail with the new logic

### Key Changes

- **Removed implicit session generation**: Eliminated automatic session creation that was happening behind the scenes
  - Make all test sessions explicit. `subset` and `record test` always require `--build` and `--session`.
- **Updated build command**: Renamed `record session --name` to `record session --session` for consistency
- **Updated build command**: Renamed `record build --name` to `record build --build` for consistency

- **Simplified session management**: Streamlined the session handling logic in `smart_tests/commands/helper.py` (162 lines removed)
- **Updated test recording**: Modified `smart_tests/commands/record/tests.py` to remove session creation logic (97 lines removed)
- **Cleaned up subset generation**: Simplified subset command logic in `smart_tests/commands/subset.py` (44 lines removed)
- **Removed session utility**: Deleted `smart_tests/utils/session.py` (107 lines removed)
- **Comprehensive test updates**: Updated all test files to work with the new session management approach

### Files Modified

- Core command files: `helper.py`, `record/tests.py`, `subset.py`
- Build and session recording: `record/build.py`, `record/session.py`
- Test runner implementations: Updated 20+ test runner files
- CI/CD: Updated GitHub Actions workflows

## Test Plan

- [x] All existing tests updated and passing
- [x] E2E tests updated for new command structure
- [x] Manual testing of session creation/recording flows
- [ ] Verification of `builds/{}/test_session_names/{}` with `no_build` parameter

🤖 Generated with [Claude Code](https://claude.ai/code). Edited by @ninjinkun.

### CODE REVIEW

**Comment by ninjinkun** [comment]
Hmm the caused the following error. I should debug the server side.
404 Client Error: Couldn't find a test session for url: https://api.mercury.launchableinc.com/intake/organizations/launchableinc/workspaces/cli/builds/16070333987/test_session_names/37

File: smart_tests/commands/helper.py [hunk 1]
```diff
@@ -9,140 +8,53 @@
...
-    return read_session(saved_build_name)
+    Returns:
+        Tuple of (build_name, test_session_id)
+    """
+    import re
+    match = re.match(r"builds/([^/]+)/test_sessions/(.+)", session_id)
+    if match:
+        return match.group(1), match.group(2)
+    else:
+        # Fallback for unexpected format
```
Comment (Konboi) [inline smart_tests/commands/helper.py:56]: Q: Don't need to throw exception?

File: smart_tests/commands/record/session.py [hunk 2]
```diff
@@ -207,29 +192,30 @@ def session(
             assert build_name is not None
             sub_path = "builds/{}/test_sessions".format(build_name)
 
-        if save_session_file:
-            write_session(build_name, "{}/{}".format(sub_path, session_id))
         if print_session:
             # what we print here gets captured and passed to `--session` in
             # later commands
             typer.echo("{}/{}".format(sub_path, session_id), nl=False)
```
[the remark this answers is not in the document]
Comment (Konboi) [reply smart_tests/commands/record/session.py:198]: ditto - subset command
https://github.com/launchableinc/cli/pull/1067/files#diff-63ab4a20feb4921852d793d93ec6d53d7d1de55a0d404b801bc9815d951d8e2bR34-R37

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -73,16 +73,16 @@ jobs:
         smart-tests verify
 
         # Tell Smart Tests about the build you are producing and testing
-        smart-tests record build --name ${GITHUB_RUN_ID}
+        smart-tests record build --build ${GITHUB_RUN_ID}
 
-        smart-tests record session --build ${GITHUB_RUN_ID} --flavor os=${{ matrix.os }} --flavor python=$(cat .python-version) > session.txt
+        smart-tests record session --build ${GITHUB_RUN_ID} --session ${GITHUB_RUN_NUMBER}-${{ matrix.os | replace('.', '-') }}  --flavor os=${{ matrix.os }} --flavor python=$(cat .python-version)
```
[the remark this answers is not in the document]
Comment (Konboi) [reply .github/workflows/test.yml:78]: Actually, now that I think about it, it should be fine since we doesn't allow `/` to the session name, right?
